remote/src/main/scala/org/apache/pekko/remote/transport/netty/NettyTransport.scala

```diff
@@ -462,23 +473,15 @@ class NettyTransport(val settings: NettyTransportSettings, val system: ExtendedA
 
   override def isResponsibleFor(address: Address): Boolean = true // TODO: Add configurable subnet filtering
 
-  // TODO: This should be factored out to an async (or thread-isolated) name lookup service #2960
-  def addressToSocketAddress(addr: Address): Future[InetSocketAddress] = addr match {
-    case Address(_, _, Some(host), Some(port)) =>
-      Future { blocking { new InetSocketAddress(InetAddress.getByName(host), port) } }
-    case _ => Future.failed(new IllegalArgumentException(s"Address [$addr] does not contain host or port information."))
-  }
-
   override def listen: Future[(Address, Promise[AssociationEventListener])] = {
     @nowarn("msg=deprecated")
     val bindPort = settings.BindPortSelector
 
     for {
-      address <- addressToSocketAddress(Address("", "", settings.BindHostname, bindPort))
+      _ <- Future.unit
```

Comment:
Is there a reason why we are using `Future.unit` like this, from what I can tell you are just lifting a value into `Future` which makes the `for` comprehension a bit redundant (i.e. you would use Future.apply`/`Future.successful`/`Future.failed` instead)?

Reply:
That will change more code (4 lines more), I have done that before but then reverted.

Reply:
I personally have a preference for clearer code versus minimizing the diff, especially for something as trivial as 4 lines.

To me the `Future.unit` in a for comprehension by itself is a case of code smell, i.e. no one would write this code from a clean slate/starting point.

Also if you use something like `Future.fromTry` (which I think is the correct constructor to be used here) it would also be faster since putting inside `for` invokes a call to an executionContext

Reply:
@mdedetrich see another PR, I will rebase this pr once https://github.com/apache/incubator-pekko/pull/651 get merged.

Reply:
Ah I see what you mean now, nvm then.